manticore/platforms/linux.py

```diff
@@ -2036,26 +2050,80 @@ class SLinux(Linux):
     Builds a symbolic extension of a Linux OS
 
     :param str programs: path to ELF binary
+    :param str disasm: disassembler to be used
     :param list argv: argv not including binary
     :param list envp: environment variables
     :param tuple[str] symbolic_files: files to consider symbolic
     """
-    def __init__(self, programs, argv=None, envp=None, symbolic_files=None):
+    def __init__(self, programs, argv=None, envp=None, symbolic_files=None,
+                 disasm='capstone'):
         argv = [] if argv is None else argv
         envp = [] if envp is None else envp
         symbolic_files = [] if symbolic_files is None else symbolic_files
 
         self._constraints = ConstraintSet()
         self.random = 0
         self.symbolic_files = symbolic_files
-        super(SLinux, self).__init__(programs, argv, envp)
+        super(SLinux, self).__init__(programs,
+                                     argv=argv,
+                                     envp=envp,
+                                     disasm=disasm)
+
 
     def _mk_proc(self, arch):
         if arch in {'i386', 'armv7'}:
             mem = SMemory32(self.constraints)
         else:
             mem = SMemory64(self.constraints)
-        return CpuFactory.get_cpu(mem, arch)
+
+        if is_binja_disassembler(self.disasm):
+            return self._init_binja_cpu(mem)
+
+        cpu = CpuFactory.get_cpu(mem, arch)
+
+        # FIXME
+        arch_map = {
+            'i386': (cs.CS_ARCH_X86, cs.CS_MODE_32),
+            'amd64': (cs.CS_ARCH_X86, cs.CS_MODE_64),
+            'armv7': (cs.CS_ARCH_ARM, cs.CS_MODE_ARM)
+        }
+        arch, mode = arch_map[arch]
+        return cpu
+
+    def _init_binja_cpu(self, memory):
+        from ..core.cpu.binja import BinjaCpu
+
+        #  FIXME (theo) this will be replaced by a function that simply
+        #  loads the IL from a file
+        def init_bv():
```

Comment:
since this takes no arguments, couldn't this be in the BinjaCpu ctor? maybe an optional `view=None` parameter? then we wouldn't need this `_init_binja_cpu` method that lives in `Linux` which feels odd to me. At the call site, we can simply call the BinjaCpu ctor?